File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        merged_list_head = ListNode(-1)
        current_node = merged_list_head

        while list1 is not None and list2 is not None:
            if list1.val <= list2.val:
                current_node.next = list1
                list1 = list1.next
            else:
                current_node.next = list2
                list2 = list2.next
            
            current_node = current_node.next
            
        if list1 is not None:
            current_node.next = list1
        else:
            current_node.next = list2
        
        merged_list_head = merged_list_head.next
        return merged_list_head

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode mergedListsHead = new ListNode(-1);
		ListNode currNode = mergedListsHead;
		
		while(list1 != null && list2 != null){
			if(list1.val <= list2.val){
				currNode.next = list1;
				list1 = list1.next;
			}
			else{
				currNode.next = list2;
				list2 = list2.next;
			}
			
			currNode = currNode.next;
		}
		
		while(list1 != null){
			currNode.next = list1;
			currNode = currNode.next;
			list1 = list1.next;
		}
		
		while(list2 != null){
			currNode.next = list2;
			currNode = currNode.next;
			list2 = list2.next;
		}
		
		currNode.next = null;
		mergedListsHead = mergedListsHead.next;
		
		return mergedListsHead;
    }
}